lib/compiler.js

```diff
@@ -0,0 +1,106 @@
+                if (isResolved) return;
+                isResolved = true;
+                if (childProcess) {
+                    childProcess.removeListener("message", resolveOnWebpackCompilationComplete);
+                }
+                rejectBase();
+            }
+
+            // TODO: Read from CLI options...
+            const { watch } = true;
+
+            console.log(`Running webpack for ${config.platform}...`);
+            const envFlagNames = Object.keys(config.env).concat([config.platform.toLowerCase()]);
+
+            const snapshotEnvIndex = envFlagNames.indexOf("snapshot");
+            if (snapshotEnvIndex !== -1 && !utils.shouldSnapshot($mobileHelper, config.platform, config.bundle)) {
+                envFlagNames.splice(snapshotEnvIndex, 1);
+            }
+
+            // Adding `npm i source-map-support --save-dev` in an app will make source maps work
+            // and stack traces will point to .ts if .ts files and proper source maps exist.
+            let sourceMapSupportArgs = [];
+            let appSourceMapSupportInstallPath = pathResolve($projectData.projectDir, "node_modules", "source-map-support", "register.js");
+            let devDepSourceMapSupportInstallPath = pathResolve(__dirname, "..", "node_modules", "source-map-support", "register.js");
+            if (existsSync(appSourceMapSupportInstallPath)) {
+                sourceMapSupportArgs = ["--require", appSourceMapSupportInstallPath];
+            } else if (existsSync(devDepSourceMapSupportInstallPath)) {
+                sourceMapSupportArgs = ["--require", devDepSourceMapSupportInstallPath];
+            }
+
+            const args = [
+                "--preserve-symlinks",
+                ...sourceMapSupportArgs,
+                join($projectData.projectDir, "node_modules", "webpack", "bin", "webpack.js"),
+                "--config=webpack.config.js",
+                "--progress",
+                ... (config.watch ? ["--watch"] : []),
+                ...envFlagNames.map(item => `--env.${item}`),
+            ].filter(a => !!a);
+
+            const childProcess = spawn("node", args, {
+                // Watch opens IPC so we don't mess with the stdin/out/err.
+                // These will notify us for the webpack compilation states.
+                // Enables `childProcess.on("message", msg => ...)` kind of communication.
+                stdio: config.watch ? ["inherit", "inherit", "inherit", "ipc"] : "inherit",
+                cwd: $projectData.projectDir
+            });
+
+            function resolveOnWebpackCompilationComplete(message) {
+                if (message === "Webpack compilation complete. Watching for file changes.") {
+                    console.log("Initial webpack build done!");
+                    resolve();
+                }
+            }
+
+            if (config.watch) {
+                childProcess.on("message", resolveOnWebpackCompilationComplete);
+                if (webpackProcess) {
```

Comment:
@Mitko-Kerezov 
Probably we will not throw here, but we will figure it out when it is integrated in the CLI,